tests/vlan/test_autostate_disabled.py

```diff
@@ -0,0 +1,120 @@
+    the vlans go down, the bridge will go down and the vlan interface will go down.
+
+    For more information about autostate, see:
+      * https://www.cisco.com/c/en/us/support/docs/switches/catalyst-6500-series-switches/41141-188.html
+    """
+
+    def test_autostate_disabled(self, duthosts, enum_frontend_dut_hostname):
+        """
+        Verify vlan interface autostate is disabled on SONiC.
+        """
+
+        duthost = duthosts[enum_frontend_dut_hostname]
+        dut_hostname = duthost.hostname
+
+        # Collect DUT configuration and status
+        vlan_members_facts = duthost.get_running_config_facts().get('VLAN_MEMBER')
+        if vlan_members_facts is None:
+            pytest.skip('No vlan available on DUT {hostname}'.format(hostname=dut_hostname))
+        ifs_status = self.get_interface_status(duthost)
+        ip_ifs = duthost.show_ip_interface()['ansible_facts']['ip_interfaces']
+
+        # Find out all vlans which meet the following requirements:
+        #   1. The oper_state of vlan interface is 'up'
+        #   2. The oper_state of at least one member in the vlan is 'up'
+        vlan_available = []
+        for vlan in vlan_members_facts:
+            if ip_ifs.get(vlan, {}).get('oper_state') == 'up':
+                for member in vlan_members_facts[vlan]:
+                    if ifs_status.get(member, {}).get('oper') == 'up':
+                        vlan_available.append(vlan)
+                        break
+        if len(vlan_available) == 0:
+            pytest.skip('No vlan available on DUT {hostname}'.format(hostname=dut_hostname))
+
+        # Pick a vlan for test
+        vlan = vlan_available[0]
+        vlan_members = vlan_members_facts[vlan].keys()
+
+        # Shutdown all the members in vlan.
+        res = duthost.shutdown_multiple(vlan_members)
+        if not res.is_successful:
+            self.restore_interface_admin_state(duthost, ifs_status)
+            pytest.fail('shutdown "{vlan_members}" in {vlan} failed'.format(vlan_members=vlan_members, vlan=vlan))
+        logging.info('waiting for "{vlan_members}" shutdown in {vlan}'.format(vlan_members=vlan_members, vlan=vlan))
+        if not wait_until(60, 5, 0, self.check_interface_oper_state, duthost, vlan_members, "down"):
+            self.restore_interface_admin_state(duthost, ifs_status)
+            pytest.fail('shutdown "{vlan_members}" in {vlan} failed'.format(vlan_members=vlan_members, vlan=vlan))
+
+        # Check whether the oper_state of vlan interface is changed as expected.
+        ip_ifs = duthost.show_ip_interface()['ansible_facts']['ip_interfaces']
+        if len(vlan_available) > 1:
+            # If more than one vlan comply with the above test requirements, then there are members in other vlans
+            # that are still up. Therefore, the bridge is still up, and vlan interface should be up.
+            if ip_ifs.get(vlan, {}).get('oper_state') != "up":
```

Comment:
Let's use `pytest_assert` directly

Reply:
Thanks, I have fixed this in latest commit.